core/kazoo_amqp/src/gen_listener.erl

```diff
@@ -582,14 +596,18 @@ maybe_remove_binding(_BP, _B, _P, _Q) -> 'true'.
 %% @end
 %%--------------------------------------------------------------------
 -spec handle_info(any(), state()) -> handle_info_ret().
-handle_info({#'basic.deliver'{}=BD, #amqp_msg{props=#'P_basic'{content_type=CT}
+handle_info({#'basic.deliver'{}=BD, #amqp_msg{props=#'P_basic'{content_type=CT}=Basic
                                              ,payload=Payload
                                              }}
-           ,#state{params=Params}=State) ->
+           ,#state{params=Params, auto_ack=AutoAck}=State) ->
+    _ = case AutoAck of
```

Comment:
Remind me why we need to track `AutoAck` here when we default to `no_ack` [here](https://github.com/2600hz/kazoo/blob/master/core/kazoo_amqp/src/amqp_util.erl#L1111)?

Reply:
no_ack defaults to true, which mean no ack needed. because we override the setting if `auto_ack` is provided, we need to `ack` the delivery here on client behalf.